swift-sdk/Internal/IterableCoreDataPersistence.swift

```diff
@@ -21,143 +21,152 @@ enum PersistenceConst {
-            return nil
-        }
-        ITBInfo("DB Bundle url: \(url)")
-        return NSManagedObjectModel(contentsOf: url)
-    }
-    
-    private static func dataModelUrl(fromBundles bundles: [Bundle]) -> URL? {
-        bundles.lazy.compactMap(dataModelUrl(fromBundle:)).first
-    }
-    
-    private static func dataModelUrl(fromBundle bundle: Bundle) -> URL? {
-        ResourceHelper.url(forResource: PersistenceConst.dataModelFileName,
-                           withExtension: PersistenceConst.dataModelExtension,
-                           fromBundle: bundle)
+        viewContext.automaticallyMergesChangesFromParent = true
+        viewContext.mergePolicy = NSMergePolicy(merge: NSMergePolicyType.mergeByPropertyStoreTrumpMergePolicyType)
     }
 }
 
-struct CoreDataPersistenceContextProvider: IterablePersistenceContextProvider {
-    init?(dateProvider: DateProviderProtocol = SystemDateProvider()) {
-        guard let persistentContainer = PersistentContainer.initialize() else {
-            return nil
-        }
+final class CoreDataPersistenceContextProvider: IterablePersistenceContextProvider {
+    init(
+        dateProvider: DateProviderProtocol = SystemDateProvider(),
+        persistentContainer: NSPersistentContainer = PersistentContainer()
+    ) {
         self.persistentContainer = persistentContainer
         self.dateProvider = dateProvider
     }
-    
+
     func newBackgroundContext() -> IterablePersistenceContext {
+        if !isStoreLoaded {
+            isStoreLoaded = loadStore(into: persistentContainer)
+        }
         return CoreDataPersistenceContext(managedObjectContext: persistentContainer.newBackgroundContext(), dateProvider: dateProvider)
     }
-    
+
     func mainQueueContext() -> IterablePersistenceContext {
+        if !isStoreLoaded {
+            isStoreLoaded = loadStore(into: persistentContainer)
+        }
         return CoreDataPersistenceContext(managedObjectContext: persistentContainer.viewContext, dateProvider: dateProvider)
     }
-    
-    private let persistentContainer: PersistentContainer
+
+    private let persistentContainer: NSPersistentContainer
     private let dateProvider: DateProviderProtocol
+    private var isStoreLoaded = false
+
+    /// Loads the persistent container synchronously so we can easily capture loading errors.
+    private func loadStore(into container: NSPersistentContainer) -> Bool {
+        if let descriptor = container.persistentStoreDescriptions.first {
+            descriptor.shouldAddStoreAsynchronously = false
+        }
+
+        // This closure runs synchronously because of the settings above
+        var loadError: (any Error)?
+        container.loadPersistentStores { _, error in
+            loadError = error
+        }
+
+        if let error = loadError {
+            ITBError("Failed to load Iterable's store. \(error.localizedDescription)")
+            return false
+        }
```

Comment:
Not change needed, but just calling out that this is probably an error in Swift 6 language mode. You likely need to do `nonisolated(unsafe)` on `leadError` to fix (since we know through static analysis this will never be a concurrency issue in this context).